cmd/okctl/delete.go

```diff
@@ -125,13 +124,18 @@ including VPC, this is a highly destructive operation.`,
 
 			formatErr := o.ErrorFormatter(fmt.Sprintf("delete cluster %s", opts.Environment), userDir)
 
-			err = services.Domain.DeletePrimaryHostedZone(o.Ctx, client.DeletePrimaryHostedZoneOpts{
+			err = services.Domain.DeletePrimaryHostedZone(o.Ctx, o.CloudProvider, client.DeletePrimaryHostedZoneOpts{
 				ID: id,
 			})
 			if err != nil {
 				return formatErr(err)
 			}
 
+			err = services.IdentityManager.DeleteIdentityPool(o.Ctx, o.CloudProvider, id)
```

Comment:
See another comment, but `CloudProvider` should not be injected here, but rather into the corresponding layer in the service.